SnapshotReplicationTest.shouldReceiveNewSnapshotsOnRejoin is flaky

**Description**

It sometimes happens that there are more snapshots than expected, even though the snapshot interval is 5 minutes, these snapshots (from the logged timestamp) seem to be taken with smaller intervals - it is a controlled clock however, so we should investigate.

NOTE: I've seen this happen with other tests in that suite, so the same issue might be related to flakiness of other tests.

<details>
<summary>Sample output</summary>
<pre>
Failures: 

[ERROR]   SnapshotReplicationTest.shouldReceiveNewSnapshotsOnRejoin:120 

Expecting:

 <{"2-1-1584316799462/000091.log"=2794790440L, "2-1-1584316799462/000092.sst"=3762184780L, "2-1-1584316799462/000093.sst"=2073560521L, "2-1-1584316799462/000095.sst"=2902222425L, "2-1-1584316799462/000096.sst"=3165565020L, "2-1-1584316799462/000097.sst"=3946582159L, "2-1-1584316799462/000098.sst"=400477988L, "2-1-1584316799462/000099.sst"=30820173L, "2-1-1584316799462/000100.sst"=1959921764L, "2-1-1584316799462/000101.sst"=3306696654L, "2-1-1584316799462/CURRENT"=2380087979L, "2-1-1584316799462/MANIFEST-000004"=4035880576L, "2-1-1584316799462/OPTIONS-000090"=3042895128L, "2-1-1584316808269/000092.sst"=3762184780L, "2-1-1584316808269/000093.sst"=2073560521L, "2-1-1584316808269/000095.sst"=2902222425L, "2-1-1584316808269/000096.sst"=3165565020L, "2-1-1584316808269/000097.sst"=3946582159L, "2-1-1584316808269/000098.sst"=400477988L, "2-1-1584316808269/000099.sst"=30820173L, "2-1-1584316808269/000100.sst"=1959921764L, "2-1-1584316808269/000101.sst"=3306696654L, "2-1-1584316808269/000103.sst"=1095198306L, "2-1-1584316808269/000108.log"=874173147L, "2-1-1584316808269/000109.sst"=35270537L, "2-1-1584316808269/000110.sst"=1968252910L, "2-1-1584316808269/000111.sst"=1199288592L, "2-1-1584316808269/000112.sst"=1337337557L, "2-1-1584316808269/000113.sst"=548214779L, "2-1-1584316808269/CURRENT"=1352198163L, "2-1-1584316808269/MANIFEST-000104"=1667902739L, "2-1-1584316808269/OPTIONS-000107"=2111876784L}>

to contain:

 <[2-1-1584316808269/000096.sst=3165565020,

    2-1-1584316808269/000103.sst=1095198306,

    2-1-1584316808269/000112.sst=1337337557,

    2-1-1584316799462/000091.log=2794790440,

    2-1-1584316802297/000095.sst=2902222425,

    2-1-1584316808269/000095.sst=2902222425,

    2-1-1584316799462/000096.sst=3165565020,

    2-1-1584316799462/000092.sst=3762184780,

    2-1-1584316799462/000095.sst=2902222425,

    2-1-1584316799462/OPTIONS-000090=3042895128,

    2-1-1584316808269/000099.sst=30820173,

    2-1-1584316808269/CURRENT=1352198163,

    2-1-1584316808269/000110.sst=1968252910,

    2-1-1584316808269/000100.sst=1959921764,

    2-1-1584316802297/000099.sst=30820173,

    2-1-1584316808269/000113.sst=548214779,

    2-1-1584316802297/OPTIONS-000090=3042895128,

    2-1-1584316808269/000109.sst=35270537,

    2-1-1584316802297/CURRENT=2380087979,

    2-1-1584316802297/000098.sst=400477988,

    2-1-1584316808269/000098.sst=400477988,

    2-1-1584316799462/MANIFEST-000004=4035880576,

    2-1-1584316808269/000101.sst=3306696654,

    2-1-1584316802297/000101.sst=3306696654,

    2-1-1584316799462/000099.sst=30820173,

    2-1-1584316802297/000100.sst=1959921764,

    2-1-1584316799462/000101.sst=3306696654,

    2-1-1584316799462/CURRENT=2380087979,

    2-1-1584316808269/000093.sst=2073560521,

    2-1-1584316802297/000093.sst=2073560521,

    2-1-1584316808269/000097.sst=3946582159,

    2-1-1584316802297/000102.sst=764452740,

    2-1-1584316802297/000097.sst=3946582159,

    2-1-1584316799462/000098.sst=400477988,

    2-1-1584316799462/000100.sst=1959921764,

    2-1-1584316808269/OPTIONS-000107=2111876784,

    2-1-1584316808269/MANIFEST-000104=1667902739,

    2-1-1584316808269/000111.sst=1199288592,

    2-1-1584316799462/000097.sst=3946582159,

    2-1-1584316802297/000096.sst=3165565020,

    2-1-1584316799462/000093.sst=2073560521,

    2-1-1584316808269/000108.log=874173147,

    2-1-1584316802297/MANIFEST-000004=1466873043,

    2-1-1584316808269/000092.sst=3762184780,

    2-1-1584316802297/000092.sst=3762184780]>

but could not find:

 <[2-1-1584316802297/000095.sst=2902222425,

    2-1-1584316802297/000099.sst=30820173,

    2-1-1584316802297/OPTIONS-000090=3042895128,

    2-1-1584316802297/CURRENT=2380087979,

    2-1-1584316802297/000098.sst=400477988,

    2-1-1584316802297/000101.sst=3306696654,

    2-1-1584316802297/000100.sst=1959921764,

    2-1-1584316802297/000093.sst=2073560521,

    2-1-1584316802297/000102.sst=764452740,

    2-1-1584316802297/000097.sst=3946582159,

    2-1-1584316802297/000096.sst=3165565020,

    2-1-1584316802297/MANIFEST-000004=1466873043,

    2-1-1584316802297/000092.sst=3762184780]>

</pre>
</details>

--- a/qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SnapshotReplicationTest.java
+++ b/qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SnapshotReplicationTest.java
@@ -77,9 +77,20 @@ public final class SnapshotReplicationTest {
   public void shouldReceiveLatestSnapshotOnRejoin() {
     // given
     final var leaderNodeId = clusteringRule.getLeaderForPartition(1).getNodeId();
-    final var otherBrokers = clusteringRule.getOtherBrokerObjects(leaderNodeId);
-    final var firstFollowerId = otherBrokers.get(0).getConfig().getCluster().getNodeId();
-    final var secondFollowerId = otherBrokers.get(1).getConfig().getCluster().getNodeId();
+    final var followers =
+        clusteringRule.getOtherBrokerObjects(leaderNodeId).stream()
+            .map(b -> b.getConfig().getCluster().getNodeId())
+            .collect(Collectors.toList());
+
+    // Temp fix to make sure we don't restart broker-0 due to
+    // https://github.com/zeebe-io/atomix/issues/208
+    final var firstFollowerId =
+        followers.stream().filter(nodeId -> !nodeId.equals(0)).findFirst().orElseThrow();
+    final var secondFollowerId =
+        followers.stream()
+            .filter(nodeId -> nodeId.equals(firstFollowerId))
+            .findFirst()
+            .orElseThrow();
 
     // when - snapshot
     clusteringRule.stopBroker(firstFollowerId);
@@ -99,9 +110,20 @@ public final class SnapshotReplicationTest {
   public void shouldReceiveNewSnapshotsOnRejoin() {
     // given
     final var leaderNodeId = clusteringRule.getLeaderForPartition(1).getNodeId();
-    final var otherBrokers = clusteringRule.getOtherBrokerObjects(leaderNodeId);
-    final var firstFollowerId = otherBrokers.get(0).getConfig().getCluster().getNodeId();
-    final var secondFollowerId = otherBrokers.get(1).getConfig().getCluster().getNodeId();
+    final var followers =
+        clusteringRule.getOtherBrokerObjects(leaderNodeId).stream()
+            .map(b -> b.getConfig().getCluster().getNodeId())
+            .collect(Collectors.toList());
+
+    // Temp fix to make sure we don't restart broker-0 due to
+    // https://github.com/zeebe-io/atomix/issues/208
+    final var firstFollowerId =
+        followers.stream().filter(nodeId -> !nodeId.equals(0)).findFirst().orElseThrow();
+    final var secondFollowerId =
+        followers.stream()
+            .filter(nodeId -> nodeId.equals(firstFollowerId))
+            .findFirst()
+            .orElseThrow();
 
     // when - snapshot
     clusteringRule.stopBroker(firstFollowerId);

